cpp/modmesh/universe/World.hpp

```diff
@@ -106,9 +106,9 @@ class World
     {
         m_segments->append(segment);
     }
-    void add_segment(value_type x0, value_type y0, value_type z0, value_type x1, value_type y1, value_type z1)
+    void add_segment(point_type const & p0, point_type const & p1)
```

Comment:
It's too confusing to use coordinate values to add segments.  No one can tell if the argument order is by point or by coordinate axis.  Change to use points instead.